protocol/contracts/v2/strategies/RamosStrategy.sol

```diff
@@ -0,0 +1,181 @@
+    function borrowAndAddLiquidity(uint256 _amount, IBalancerVault.JoinPoolRequest memory _requestData) external onlyElevatedAccess {
+        // Borrow the DAI and send it to RAMOS. This will also mint `dUSD` debt.
+        treasuryReservesVault.borrow(_amount, address(ramos));
+        emit BorrowAndAddLiquidity(_amount);
+
+        // Add liquidity
+        ramos.addLiquidity(_requestData);
+    }
+
+    /// @notice Get the quote used to remove liquidity
+    /// @dev Since this is not the view function, this should be called with `callStatic`
+    function proportionalRemoveLiquidityQuote(
+        uint256 _bptAmount,
+        uint256 _slippageBps
+    ) public returns (
+        uint256 expectedTempleAmount,
+        uint256 expectedStablesAmount,
+        uint256 minTempleAmount,
+        uint256 minStablesAmount,
+        IBalancerVault.ExitPoolRequest memory requestData
+    ) {
+        return ramos.proportionalRemoveLiquidityQuote(_bptAmount, _slippageBps);
+    }
+
+    /**
+     * @notice Remove liquidity and repay debt back to the Treasury Reserves
+     * The stables from the removed liquidity are sent from RAMOS to this address to repay debt
+     */
+    function removeLiquidityAndRepay(IBalancerVault.ExitPoolRequest memory _requestData, uint256 _bptAmount) external onlyElevatedAccess {
+        // Remove liquidity
+        ramos.removeLiquidity(_requestData, _bptAmount, address(this));
+
+        // Repay debt back to the Treasury Reserves
+        uint256 stableBalance = stableToken.balanceOf(address(this));
+
+        treasuryReservesVault.repay(stableBalance, address(this));
+        emit RemoveLiquidityAndRepay(stableBalance);
+    }
+
+    struct PopulateShutdownParams {
+        uint256 slippageBps;
+    }
+
+    struct ShutdownParams {
+        IBalancerVault.ExitPoolRequest requestData;
+        uint256 bptAmount;
+    }
+
+    /**
+     * @notice Populate data to automatically shutdown.
+     * This gets a quote to unstake all BPT and liquidate into stables.
+     * @param populateParamsData abi encoded data of struct `PopulateShutdownParams`
+     * @return shutdownData abi encoded data of struct `ShutdownParams`
+     */
+    function populateShutdownData(
+        bytes memory populateParamsData
+    ) external virtual override returns (
+        bytes memory shutdownData
+    ) {
+        (PopulateShutdownParams memory populateParams) = abi.decode(populateParamsData, (PopulateShutdownParams));
+        ShutdownParams memory shutdownParams;
+
+        (shutdownParams.bptAmount,,) = ramos.positions();
+        (,,,,shutdownParams.requestData) = proportionalRemoveLiquidityQuote(shutdownParams.bptAmount, populateParams.slippageBps);
+        shutdownData = abi.encode(shutdownParams);
+    }
+
+    /**
+     * @notice Shutdown the strategy.
+     * First unstake all BPT and liquidate into stables, and then repay the stables to the TRV.
+     * @param shutdownData abi encoded data of struct `PopulateShutdownParams`
```

Comment:
fix up comment

Reply:
Fixed in https://github.com/TempleDAO/temple/pull/814/commits/d5b1bc654ab8a87badb80eb957290150f4ba2c8b